run_project_tests.py

```diff
@@ -1060,17 +1050,12 @@ def should_skip_rust(backend: Backend) -> bool:
         return True
     if backend is not Backend.ninja:
         return True
-    if mesonlib.is_windows():
-        if has_broken_rustc():
-            return True
-    return False
+    return bool(mesonlib.is_windows() and has_broken_rustc())
```

Comment:
ditto

Reply:
This function is promised to return a `bool`.

Ruff lints superfast (1/3 of a secord for the entire CPython codebase) so it does not assume or validate that the retrun type of has_broken_rustc() `bool` or not.

Reply:
And both of the called functions promise to return a bool (and also do return a bool). I understand that ruff is not a static type checker, but if it's smart enough to notice that this function promises returns a bool, but not that the two called functions also promise to return a bool, that's surprising.

Reply:
I suspect ruff is actually a lot "dumber" (read: simple) and simply rewrites all code of the form:

```python
if cond():
    return BOOL
return OTHERBOOL
```
to:
```python
return BOOL if cond() else OTHERBOOL
```

And before that, it also rewrote:

```python
if cond1():
    if cond2():
        statement
```

to:
```python
if cond1() and cond2():
    statement
```

Ruff is doing code structure analysis, not type analysis, and not even return type analysis. It knows how to handle  builtin types, so that it can see the equivalence of `ret = bool(X)` and `if X: ret = True`.